sphinx/source/docs/user_guide/webgl.rst

```diff
@@ -23,56 +25,75 @@ To enable WebGL in Bokeh, set the plot's ``output_backend`` property to
 
     p = figure(output_backend="webgl")
 
-Support
--------
+The WebGL output backend only supports a :ref:`subset of Bokeh's glyphs
+<userguide_webgl_supported_glyphs>`. When you enable WebGL, you are requesting
+that WebGL rendering is used if available. Glyphs that are not available in the
+WebGL output backend are rendered with the default canvas backend.
 
-Bokeh's WebGL support covers a subset of glyphs. This includes the :func:`~bokeh.plotting.figure.line`
-glyph, and most markers:
+.. _userguide_webgl_supported_glyphs:
 
-* :func:`~bokeh.plotting.figure.asterisk`
+Supported glyphs
+----------------
+
+Bokeh's WebGL support covers the following fixed-shape glyphs:
+
+* :func:`~bokeh.plotting.figure.block`
 
 * :func:`~bokeh.plotting.figure.circle`
 
-* :func:`~bokeh.plotting.figure.circle_cross`
+* :func:`~bokeh.plotting.figure.hbar`
 
-* :func:`~bokeh.plotting.figure.circle_x`
+* :func:`~bokeh.plotting.figure.hex_tile`
 
-* :func:`~bokeh.plotting.figure.cross`
+* :func:`~bokeh.plotting.figure.quad`
 
-* :func:`~bokeh.plotting.figure.diamond`
+* :func:`~bokeh.plotting.figure.rect`
 
-* :func:`~bokeh.plotting.figure.diamond_cross`
+* :func:`~bokeh.plotting.figure.scatter`
 
-* :func:`~bokeh.plotting.figure.hex`
+* :func:`~bokeh.plotting.figure.vbar`
 
-* :func:`~bokeh.plotting.figure.inverted_triangle`
+WebGL support for these fixed-shape glyphs covers the following properties:
 
-* :func:`~bokeh.plotting.figure.square`
+* all :ref:`fill properties <userguide_styling_fill_properties>`
+* all :ref:`line properties <userguide_styling_line_properties>` except for
+  dashed lines (which will be ignored)
+* all :ref:`hatch properties <userguide_styling_hatch_properties>` except for
+  hatch images using the ``hatch_extra`` property (which will be ignored).
 
-* :func:`~bokeh.plotting.figure.square_cross`
+There is also full WebGL support for :func:`~bokeh.plotting.figure.line`
+glyphs, although the appearance of dashed lines with round and square end caps
+needs improvement.
 
-* :func:`~bokeh.plotting.figure.square_x`
+If you enable Bokeh's webGL output backend, WebGL will be used whenever
+supported by a browser. This includes output in :ref:`Jupyter notebooks or
+Jupyter lab <userguide_jupyter>` and when :ref:`exporting Bokeh plots to PNG
+<userguide_export>` if the underlying browser (including headless browsers)
+supports WebGL. In case a browser does not support WebGL, Bokeh automatically
+falls back to the standard canvas output backend.
 
-* :func:`~bokeh.plotting.figure.star`
 
-* :func:`~bokeh.plotting.figure.triangle`
+When to use WebGL
+-----------------
 
-* :func:`~bokeh.plotting.figure.x`
+A general rule of thumb is that Bokeh's default canvas output backend works well
+if you are rendering fewer than 10,000 markers or points. For plots with more
+than 10,000 markers or points, WebGL will usually give noticeably better
```

Comment:
Where did the 10k value come from? It seems low IMO. If I up the `color_scatter.py` example to N=50000 it still performs fairly responsively to me, even with LOD completely disabled.

Reply:
It comes from my dev system, which probably has better hardware acceleration than a typical dev system?

Reply:
Maybe we should be less specific then. Fewer than 10k = canvas is fine, greater than 100k = webgl better, in between it depends on your hardware, etc so you need to try it out yourself to see what is best.

Reply:
This sounds good to me, although I'd also be fine with just a slightly higher value, say 25k? Whichever you think is better.

Reply:
I've opted for 25k.

Reply:
When talking about canvas' performance, one should also consider what Nk data points statement actually refers to. Canvas will be perfectly happy to render a lot of markers, as long as one doesn't change visuals. If every marker has a different color, then performance will degrade significantly. This is the area where webgl backend is particularly strong at.

Reply:
I was testing at this level with `color_scatter.py` I think 25k is a reasonable compromise heuristic. Canvas can certainly do better in some cases.